File: 프로그래머스/2/42860. 조이스틱/조이스틱.java
public class Solution {
    public int solution(String name) {
        int length = name.length();
        int answer = 0;
        int minMove = length - 1;

        for (int i = 0; i < length; i++) {
            char c = name.charAt(i);
            int diffA = c - 'A';
            int diffZ = 'Z' - c + 1;
            answer += Math.min(diffA, diffZ);

            int next = i + 1;
            while (next < length && name.charAt(next) == 'A') {
                next++;
            }
            minMove = Math.min(minMove, i + length - next + Math.min(i, length - next));
        }

        answer += minMove;
        return answer;
    }

}

File: 프로그래머스/2/42860. 조이스틱/조이스틱.py
def solution(name):
    answer = 0
    n=len(name)
    move=n-1
    
    for c in range (n):
        a=abs(ord('A')-ord(name[c]))
        b=abs(ord('Z')-ord(name[c]))+1
        
        answer+=min(a,b)
        
        next=c+1
        while next<n and name[next]=='A':
            next+=1
            
        move=min(move,c*2+n-next)
        move=min(move,(n-next)*2+c)
        
    return answer+move